sklearn/utils/_param_validation.py

```diff
@@ -478,7 +478,7 @@ def _check_params(self):
             )
 
     def __contains__(self, val):
-        if np.isnan(val):
+        if not isinstance(val, Integral) and np.isnan(val):
```

Comment:
Hmm, this looks like an issue with `is_scalar_nan` as well. Can we fix `is_scalar_nan` and use it here?

https://github.com/scikit-learn/scikit-learn/blob/4163b32ca929972292e2b196fcc70a040ea8223f/sklearn/utils/__init__.py#L1087

```python
def is_scalar_nan(val):
    return (
        not isinstance(x, Integral) and isinstance(x, numbers.Real) and math.isnan(x)
    )
```

Reply:
I've fixed `is_scalar_nan`, but using it in _param_validation (`from . import is_scalar_nan`) gives a circular import error. Is there an easy workaround?